src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionPool.cs

```diff
@@ -1376,9 +1376,17 @@ private void PoolCreateRequest(object state)
                                 {
                                     while (NeedToReplenish)
                                     {
-                                        // Don't specify any user options because there is no outer connection associated with the new connection
-                                        newObj = CreateObject(owningObject: null, userOptions: null, oldConnection: null);
-
+                                        try
+                                        {
+                                            // Don't specify any user options because there is no outer connection associated with the new connection
+                                            newObj = CreateObject(owningObject: null, userOptions: null, oldConnection: null);
+                                        }
+                                        catch(Exception)
+                                        {
+                                            // Catch all the exceptions occuring during CreateObject so that they 
+                                            // don't emerge as unhandled on the thread pool and don't crash applications
```

Comment:
It'd be good to add why it's ok that we eat them... from your PR description, it sounds like there's some other backchannel through which they're passed up to the originating code?

Reply:
Sounds good.
What happens is that CreateObject handles the exception at https://github.com/dotnet/corefx/blob/51bdecea4f9d6020680d9ffde8ae1874e45322ba/src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionPool.cs#L721

And then it sets the ErrorEvent. The caller of the connection pool is signalled on the connection pool about the ErrorEvent being set and it knows that an exception was thrown. The caller of Connection.Open() ultimately gets the same exception that they were receiving on .Net Framework. It's just that the exception doesn't escape on to the threadpool